modules/generic-base/src/main/scala/pureconfig/generic/CoproductHint.scala

```diff
@@ -57,6 +36,19 @@ trait CoproductHint[T] {
    */
   def noOptionFound(cur: ConfigCursor): ConfigReaderFailures =
     ConfigReaderFailures(cur.failureFor(NoValidCoproductChoiceFound(cur.value)))
+
+  /**
+   * Given the current value of the coproduct option and the writer currently in scope for it, return the `ConfigValue`
+   * with its representation, possibly encoded with disambiguation information.
+   *
+   * @param writer the `ConfigWriter` in scope for type `A`
+   * @param name the name of the class or coproduct option
+   * @param value the value of the coproduct option
+   * @tparam A the type of the class or coproduct option
+   * @return the config for the sealed family or coproduct wrapped in a `Right`, or a `Left` with the failure if some
+   *         error occurred.
+   */
+  def to[A <: T](writer: ConfigWriter[A], name: String, value: A): ConfigReader.Result[ConfigValue]
```

Comment:
> I know this was like this before, but why do we return a `ConfigReader.Result` here if the write path does not handle errors? Since we are changing the API, shouldn't we drop `Result` here and throw exceptions on hints if needed? As it is it may be misleading to users.

I dropped the `Result` and started throwing `CoproductHintException`s in 977aacf323b3d687bd458943595a01d97c8cc7b9. A `CoproductHintException` wraps a single `FailureReason`, which is enough for our failure scenarios in hints.